Automattic/wp-calypso
# Framework: Migrate "lib/accept" to single test runner

Related: #3942

This pull request seeks to migrate tests for `lib/accept` to the single test runner. It also includes revisions to the `i18n` mixin to better accommodate test contexts.

__Implementation notes:__

`translate` stubbing has been swapped with the recently-added [`localize` higher-order component](https://github.com/Automattic/wp-calypso/tree/master/client/lib/mixins/i18n/localize). This alone was not sufficient to replace the existing `translate` stub, as the `i18n` module errors when trying to call `translate` before Jed has been initialized. To accommodate this, if a translate attempt is made before Jed is initialized, an empty Jed instance is created on-demand (23aed07). This should not affect existing browser usage as `i18n` is [initialized during boot](https://github.com/Automattic/wp-calypso/blob/5a1c357e6b56515ad69e740abaae45c506825e09/client/boot/index.js#L65), but will simplify testing (avoiding need for explicit i18n initialization).

__Testing instructions:__

Verify both that Mocha tests pass, and that no regressions occur in `accept` usage. An example of `accept` is the prompt which occurs when trying to delete media from your [post editor media library](http://calypso.localhost:3000/post).

/cc @rralian , @gziolo
